src/First/README.md

```diff
@@ -0,0 +1,451 @@
+# First
+
+```haskell
+First a = First (Maybe a)
+```
+
+`First` is a `Monoid` that will always return the first, non-empty value when
+(2) `First` instances are combined. `First` is able to be a `Monoid` because
+it implements a `Maybe` under the hood. The use of the `Maybe` allows for an
+empty to be represented with a `Nothing`.
+
+`First` can be constructed with either a value or a `Maybe` instance. Any value
+passed to the constructor will be wrapped in a `Just` to represent a non-empty
+instance of `First`. Any `Maybe` passed to the constructor will be lifted as
+is, allowing the ability to "choose" an value based on some disjunction.
+
+While most `Monoid`s only provide a [`valueOf`](#valueof) function used for
+extraction, `Last` takes advantage of its underlying `Maybe` to provide an
+additional [`option`](#option) method. Using [`valueOf`](#valueof) will extract
+the underlying `Maybe`, while [`option`](#option) will extract the underlying
+value in the `Maybe`, using the provided default value when the underlying
+`Maybe` is a `Nothing` instance.
+
+```javascript
+const First = require('crocks/First')
+
+const and = require('crocks/logic/and')
+const isNumber = require('crocks/predicates/isNumber')
+const mconcatMap = require('crocks/helpers/mconcatMap')
+const safe = require('crocks/Maybe/safe')
+
+// isEven :: Number -> Boolean
+const isEven =
+  x => !(x % 2)
+
+// isValid :: a -> Boolean
+const isValid =
+  and(isNumber, isEven)
+
+// chooseFirst :: [ * ] -> First Number
+const chooseFirst =
+  mconcatMap(First, safe(isValid))
+
+chooseFirst([ 21, 45, 2, 22, 19 ])
+  .valueOf(0)
+//=> Just 2
+
+chooseFirst([ 'a', 'b', 'c' ])
+  .option('')
+//=> ""
+```
+
+## Implements
+
+`Semigroup`, `Monoid`
+
+## Constructor Methods
+
+#### empty
+
+```haskell
+First.empty :: () -> First a
+```
+
+`empty` provides the identity for the `Monoid` in that when the value it
+provides is `concat`ed to any other value, it will return the other value. In
+the case of `First` the result of `empty` is `Nothing`. `empty` is available
+on both the Constructor and the Instance for convenience.
+
+```javascript
+const First = require('crocks/First')
+const { empty } = First
+
+First.empty()
+//=> First( Nothing )
+
+First(3)
+  .concat(empty())
+//=> First( Just 3 )
+
+empty()
+  .concat(First(3))
+//=> First( Just 3 )
+```
+
+## Instance Methods
+
+#### concat
+
+```haskell
+First a ~> First a -> First a
```

Comment:
I might be brainfreezing – what does the `~>` (tilde-arrow) signify?

Reply:
That signifies an instance in which the given method is executed on, so this reads:

When called on an Instance of `First a`, give me a `First a` and I will give back a `First a`.